skimage/morphology/extrema.py

```diff
@@ -207,152 +211,365 @@ def h_minima(image, h, selem=None):
 
-    We create an image (quadratic function with a maximum in the center and
-    4 additional constant maxima.
-    The heights of the maxima are: 1, 21, 41, 61, 81, 101
+    # Copy old image into new space
+    original_slice = tuple(slice(1, -1) for _ in range(image.ndim))
+    new_image[original_slice] = image
+    # and set the edge values
+    _set_edge_values_inplace(new_image, value)
 
-    >>> w = 10
-    >>> x, y = np.mgrid[0:w,0:w]
-    >>> f = 20 - 0.2*((x - w/2)**2 + (y-w/2)**2)
-    >>> f[2:4,2:4] = 40; f[2:4,7:9] = 60; f[7:9,2:4] = 80; f[7:9,7:9] = 100
-    >>> f = f.astype(np.int)
+    return new_image
+
+
+def _resolve_neighborhood(selem, connectivity, ndim):
+    """Validate or create structuring element for use in `local_maxima`.
 
-    We can calculate all local maxima:
+    Depending on the values of `connectivity` and `selem` this function
+    either creates a new structuring element (`selem` is None) using
+    `connectivity` or validates the given structuring element (`selem` is not
+    None).
 
-    >>> maxima = extrema.local_maxima(f)
+    Parameters
+    ----------
+    selem : array-like or None
+        The structuring element to validate. See same argument in
+        `local_maxima`.
+    connectivity : int or None
+        A number used to determine the neighborhood of each evaluated pixel.
+        See same argument in `local_maxima`.
+    ndim : int
+        Number of dimensions `selem` ought to have.
 
-    The resulting image will contain all 6 local maxima.
+    Returns
+    -------
+    selem : ndarray
+        Validated or new structuring element specifying the neighborhood.
     """
-    # find the minimal grey level difference
-    h = _find_min_diff(image)
-    if h == 0:
-        return np.zeros(image.shape, np.uint8)
-    if not np.issubdtype(image.dtype, np.floating):
-        h = 1
-    local_max = h_maxima(image, h, selem=selem)
-    return local_max
+    if selem is None:
+        if connectivity is None:
+            connectivity = ndim
+        selem = ndi.generate_binary_structure(ndim, connectivity)
+    else:
+        # Validate custom structured element
+        selem = np.asarray(selem, dtype=bool)
+        # Must specify neighbors for all dimensions
+        if selem.ndim != ndim:
+            raise ValueError("selem and image must have the same number of "
+                             "dimensions")
+        # Must only specify direct neighbors
+        if any(s != 3 for s in selem.shape):
+            raise ValueError("dimension size in selem is not 3")
 
+    return selem
 
-def local_minima(image, selem=None):
-    """Determine all local minima of the image.
 
-    The local minima are defined as connected sets of pixels with equal
-    grey level strictly smaller than the grey levels of all pixels in direct
-    neighborhood of the set.
+def local_maxima(image, connectivity=None, selem=None, indices=False,
+                 allow_borders=True):
```

Comment:
Even though I prefer `connectivity` as an arg, `selem` was already there in 0.13, so it needs to come first so that we don't break that interface.

Reply:
I guess this also means, that we have to keep using `selem` instead of `structure` here.

Reply:
Sadly, yes. In the future I'd love to deprecate it but not for this PR. Thanks!